### PR TITLE
EOS-24098: Configure log directory

### PR DESCRIPTION
**Pull request by selvakumaar5496**
Signed-off-by: Selvakumar <selva.k.nambi@seagate.com>

# Problem Statement
- Log directory for cortx-py-utils should be configurable

# Design
-  Read log directory from cortx.conf file

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [ ] Confirm Testing was performed with installed RPM [Y/N]:  Y

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/cortx.conf.sample [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "install_path": "/opt/seagate",
   "utils": {
+    "log_dir": "/var/log",
```
Comment (sachinpunadikar) [inline py-utils/cortx.conf.sample:4]: This should not be utils specific. The way install_path is for entire CORTX, we can have log_path

Comment (selvakumaar5496) [reply py-utils/cortx.conf.sample]: Done

File: py-utils/src/setup/utils_setup.py [hunk 2]
```diff
@@ -215,8 +215,17 @@ def process(self):
...
 
 def main(argv: dict):
-    Log.init('utils_setup', '/var/log/cortx/utils', level='INFO',
-        backup_count=5, file_size_in_mb=5)
+    from cortx.utils.conf_store import Conf
+    config_file_path = "/etc/cortx/cortx.conf"
+    Conf.load('config_file', f'yaml:///{config_file_path}', skip_reload=True)
+    # Get the log path
+    log_dir = Conf.get('config_file', 'log_dir')
+    log_path = os.path.join(log_dir, 'cortx/utils')
```
Comment (sachinpunadikar) [inline py-utils/src/setup/utils_setup.py:223]: log_path -> utils_log_path

**Comment by sachinpunadikar** [comment]
Need to get the log_dir from passed in confstore (from provisioner / template file) in mini-provisioning. If this is not provided, there should be a default value

File: py-utils/src/setup/utils_setup.py (hunk 2 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils_setup.py]: done

**Review by vastradparayya — APPROVED** [review]

File: py-utils/src/setup/utils_setup.py [hunk 3]
```diff
@@ -215,8 +215,17 @@ def process(self):
 
 
 def main(argv: dict):
-    Log.init('utils_setup', '/var/log/cortx/utils', level='INFO',
-        backup_count=5, file_size_in_mb=5)
+    from cortx.utils.conf_store import Conf
+    config_file_path = '/etc/cortx/cortx.conf'
+    Conf.load('config_file', f'yaml:///{config_file_path}', skip_reload=True)
+    # Get the log path
+    log_dir = Conf.get('config_file', 'log_dir', '/var/log')
```
Comment (sachinpunadikar) [inline py-utils/src/setup/utils_setup.py:222]: this should be read from passed in information (like populated template file)

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
please work on comment

File: py-utils/src/setup/utils_setup.py (hunk 3 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils_setup.py:222]: done in utils_setup config phase

Comment (sachinpunadikar) [reply py-utils/src/setup/utils_setup.py:222]: config phase will set the log_dir for rest of the code.
But utils_setup log will go to the default place (i.e. /var/log/..). Hence asking to get the required log directory here as well, and create log file for utils_setup in that directory.

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**